client/src/components/_common/Button/Button.jsx

```diff
@@ -55,30 +82,43 @@ const Button = ({
     buttonSizeClass = styles[`width-${size}`];
   }
 
+  const buttonLoadingClass = isLoading ? styles['loading'] : '';
+
   return (
     <button
-      className={`c-button ${buttonTypeClass} ${buttonSizeClass}`}
+      className={`
+        ${buttonRootClass}
+        ${buttonTypeClass}
+        ${buttonSizeClass}
+        ${buttonLoadingClass}
+      `}
       disabled={disabled || isLoading}
       type={attr}
       onClick={onclick}
     >
       {isLoading && (
         <LoadingSpinner
-          placement={styles['over-text']}
+          placement="inline"
           className={styles['loading-over-button']}
         />
       )}
-      <Icon
-        name={iconNameBefore}
-        className={iconNameBefore ? styles['icon--before'] : ''}
-      ></Icon>
-      <span className={isLoading ? styles['loading-text'] : ''}>
-        {children}
-      </span>
-      <Icon
-        name={iconNameAfter}
-        className={iconNameAfter ? styles['icon--after'] : ''}
-      ></Icon>
+      {iconNameBefore ? (
+        <Icon
+          name={iconNameBefore}
+          className={iconNameBefore ? styles['icon--before'] : ''}
+        ></Icon>
+      ) : (
+        ''
+      )}
+      <span className={styles['text']}>{children}</span>
+      {iconNameAfter ? (
+        <Icon
+          name={iconNameAfter}
+          className={iconNameAfter ? styles['icon--after'] : ''}
+        ></Icon>
+      ) : (
+        ''
+      )}
```

Comment:
Could be slightly simplified (could also apply to `iconNameBefore`)
```suggestion
      {iconNameAfter && (
        <Icon
          name={iconNameAfter}
          className={iconNameAfter ? styles['icon--after'] : ''}
        />
      )}
```

Reply:
Thanks. Updated both: dd360c5.